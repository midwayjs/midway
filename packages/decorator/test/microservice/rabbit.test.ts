import {
  Consumer,
  getClassMetadata,
  getPropertyDataFromClass,
  MS_CONSUMER_KEY,
  MSListenerType,
  RabbitMQListener,
  ConsumerQueuePattern,
  MS_CONSUMER_QUEUE_METADATA
} from '../../src';

@Consumer(MSListenerType.RABBITMQ)
class TestFun {

  @RabbitMQListener('tasks')
  async gotEvent() {
  }
}

@Consumer(MSListenerType.RABBITMQ, {
  exchange: 'abc'
})
class NewTestFun {

  @ConsumerQueuePattern('tasks')
  async gotEvent() {
  }
}

describe('/test/microservice/rabbit.test.ts', () => {
  it('test rabbit listener decorator', () => {
    const meta = getClassMetadata(MS_CONSUMER_KEY, TestFun);
    expect(meta).toEqual({"metadata": {}, "type": "rabbitmq"});

    const data = getPropertyDataFromClass(MS_CONSUMER_KEY, TestFun, 'gotEvent');
    expect(data[0]).toEqual({ 'propertyKey': 'gotEvent', 'queueName': 'tasks' });
  });

  it('should got method from decorator', function () {
    const meta = getClassMetadata(MS_CONSUMER_QUEUE_METADATA, NewTestFun);
    expect(meta).toMatchSnapshot();
  });
});
