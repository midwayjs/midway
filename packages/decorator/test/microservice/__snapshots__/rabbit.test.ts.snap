// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/test/microservice/rabbit.test.ts should got method from decorator 1`] = `
Array [
  Object {
    "metadata": Object {
      "pattern": "tasks",
    },
    "methodName": "gotEvent",
    "queueMethodName": "queue_pattern",
  },
]
`;
